app/controllers/admin/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class Admin::SuppliersController < Admin::BaseController
+  skip_after_action :verify_authorized, only: :index
+  skip_after_action :verify_policy_scoped, except: :index
+
+  def index
+    lead_providers = policy_scope(LeadProvider)
+    delivery_partners = policy_scope(DeliveryPartner)
+    sorted_suppliers = (lead_providers + delivery_partners).sort_by(&:name)
+    @suppliers = Kaminari.paginate_array(sorted_suppliers).page(params[:page]).per(20)
+    @page = @suppliers.current_page
+    @total_pages = @suppliers.total_pages
+  end
+
+  def new
+    if params[:continue]
+      @new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    else
+      session.delete(:new_supplier_form)
+      session.delete(:lead_provider_form)
+      @new_supplier_form = NewSupplierForm.new
+    end
+    skip_authorization
```

Comment:
I don't get why we are skipping authorization and why at the end?

Reply:
What would you authorize here? They haven't chosen lead provider vs delivery partner yet

Reply:
Could do a skip_after_action